CMakeLists.txt

```diff
@@ -723,35 +723,61 @@ ENDIF()
-  FIND_PACKAGE(CUDA REQUIRED)
-  SET(CUDA_LINK_LIBRARIES_KEYWORD PRIVATE)
-  #set(CUDA_NVCC_FLAGS
-  #  "-arch=sm_20;-Drestrict=__restrict__;-DNO_CUDA_MAIN;-O3;-use_fast_math")
-  if(CUDA_NVCC_FLAGS MATCHES "arch")
-    # User defined NVCC flags
-    MESSAGE(STATUS "Setting CUDA FLAGS=${CUDA_NVCC_FLAGS}")
-  else(CUDA_NVCC_FLAGS MATCHES "arch")
-    # Automatically set the default NVCC flags
-    SET(CUDA_NVCC_FLAGS "-Drestrict=__restrict__;-DNO_CUDA_MAIN;-std=c++14")
-    if (QMC_COMPLEX)
-      SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-DQMC_COMPLEX")
-    endif()
-    if ( CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
-      SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-g;-G")
-    else()
-      # Temporarily disable fast_math because it causes multiple test failures
-      # SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-O3;-use_fast_math")
-      SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-O3")
-    ENDIF()
-    SET( CUDA_ARCH sm_70 CACHE STRING "CUDA architecture sm_XX")
-    SET(CUDA_NVCC_FLAGS "-arch=${CUDA_ARCH};${CUDA_NVCC_FLAGS}")
-  endif(CUDA_NVCC_FLAGS MATCHES "arch")
-  INCLUDE_DIRECTORIES(${CUDA_INCLUDE_DIRS})
-  SET(HAVE_CUDA 1)
-  MESSAGE("   CUDA_NVCC_FLAGS=${CUDA_NVCC_FLAGS}")
+  IF(NOT QMC_CUDA2HIP)
+    # FindCUDA default CUDA_PROPAGATE_HOST_FLAGS to ON but we prefer OFF
+    # It happened -ffast-math from host caused numerical issue in CUDA kernels.
+    OPTION(CUDA_PROPAGATE_HOST_FLAGS "Propagate C/CXX_FLAGS and friends to the host compiler via -Xcompile" OFF)
+    FIND_PACKAGE(CUDA REQUIRED)
+    SET(CUDA_LINK_LIBRARIES_KEYWORD PRIVATE)
+    #set(CUDA_NVCC_FLAGS
+    #  "-arch=sm_20;-Drestrict=__restrict__;-DNO_CUDA_MAIN;-O3;-use_fast_math")
+    if(CUDA_NVCC_FLAGS MATCHES "arch")
+      # User defined NVCC flags
+      MESSAGE(STATUS "Setting CUDA FLAGS=${CUDA_NVCC_FLAGS}")
+    else(CUDA_NVCC_FLAGS MATCHES "arch")
+      # Automatically set the default NVCC flags
+      SET(CUDA_NVCC_FLAGS "-Drestrict=__restrict__;-DNO_CUDA_MAIN;-std=c++14")
+      if (QMC_COMPLEX)
+        SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-DQMC_COMPLEX")
+      endif()
+      if ( CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
+        SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-g;-G")
+      else()
+        # Temporarily disable fast_math because it causes multiple test failures
+        # SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-O3;-use_fast_math")
+        SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};-O3")
+      ENDIF()
+      SET( CUDA_ARCH sm_70 CACHE STRING "CUDA architecture sm_XX")
+      SET(CUDA_NVCC_FLAGS "-arch=${CUDA_ARCH};${CUDA_NVCC_FLAGS}")
+    endif(CUDA_NVCC_FLAGS MATCHES "arch")
+    INCLUDE_DIRECTORIES(${CUDA_INCLUDE_DIRS})
+    SET(HAVE_CUDA 1)
+    MESSAGE("   CUDA_NVCC_FLAGS=${CUDA_NVCC_FLAGS}")
+  ELSE()
+    SET(CMAKE_MODULE_PATH "/opt/rocm/hip/cmake;${CMAKE_MODULE_PATH}")
+    SET(CMAKE_MODULE_PATH "$ENV{ROCM_PATH}/hip/cmake;${CMAKE_MODULE_PATH}")
+    FIND_PACKAGE(HIP REQUIRED)
+    INCLUDE_DIRECTORIES("/opt/rocm/include")
```

Comment:
hard-coded path is bad. FIND_PACKAGE(HIP REQUIRED) should have setup some variables and then use them in INCLUDE_DIRECTORIES

Reply:
Without the path in `CMAKE_MODULE_PATH` CMake does not find it.
I tried to make it clean:
```
	IF(NOT DEFINED ROCM_ROOT)
        SET(ROCM_ROOT "/opt/rocm")
    ENDIF()
	LIST(APPEND CMAKE_MODULE_PATH ${ROCM_ROOT}/hip/cmake)
```